olp-cpp-sdk-dataservice-read/src/repositories/CatalogRepository.cpp

```diff
@@ -184,6 +184,96 @@ CatalogVersionResponse CatalogRepository::GetLatestVersion(
   return version_response;
 }
 
+client::CancellationToken CatalogRepository::GetLatestVersion(
+    const CatalogVersionRequest& request, CatalogVersionCallback callback) {
+  repository::CatalogCacheRepository repository(
+      catalog_, settings_.cache, settings_.default_cache_expiration);
+
+  const auto fetch_option = request.GetFetchOption();
+  // in case get version online was never called and version was not found in
+  // the cache.
+  CatalogVersionResponse version_response = {
+      {client::ErrorCode::NotFound, "Failed to find a catalog version."}};
+
+  auto func = [=, &request,
```

Comment:
This lambda is quite big. What about extracting it to a separate method or an anonymous function?
Also, the name `func` is bad because it does not describe its purpose.